### PR TITLE
sync: outline for supporting multiple ciphers used in bitwarden

### PR DESCRIPTION
**Pull request by esiqveland**
This is an idea for supporting multiple ciphers in bitw.

Just wanted to get your thoughts on this approach before I try to make it work.

See https://github.com/mvdan/bitw/issues/22

### CODE REVIEW

File: sync.go [hunk 1]
```diff
@@ -81,6 +223,10 @@ func b64decode(src []byte) ([]byte, error) {
 	return dst, nil
 }
 
+func b64decodeStr(src string) ([]byte, error) {
+	return b64decode([]byte(src))
+}
```
Comment (mvdan) [inline sync.go:228]: this is unused

File: sync.go [hunk 2]
```diff
@@ -22,14 +22,157 @@ type SyncData struct {
...
+		return s, err
+	}
+	if s.CT, err = b64decode(parts[1]); err != nil {
+		return s, err
+	}
+
+	return s, nil
+}
+
+func (i CipherStringType) String() string {
```
Comment (mvdan) [inline sync.go:148]: use `stringer` instead

**Review by mvdan — COMMENTED** [review]
Looks fine so far :) It does seem like we should support multiple ciphers.

I've left a couple of drive-by comments. My only other main worry is tests. Setting up extra real accounts to end-to-end test these ciphers seems like a lot of work for little benefit, so I think a table-based unit test to parse each cipher type would suffice.

**Comment by mvdan** [comment]
@esiqveland do you intend to finish this PR? There's some work on bitw I want to do in the coming week, so it would be ideal to get this PR merged first to avoid conflicts.

**Comment by esiqveland** [comment]
Just go ahead, I think this needs some more thought anyway. It's an idea at least for supporting more ciphers.

Also, I changed password and my account was fixed, and its now in the new format as expected by bitw current.

**Comment by mvdan** [comment]
Since I want to take a stab at https://github.com/mvdan/bitw/issues/22, I'll merge a cleaned up version of your commit to master to work on top of it.

**Pull request closed without merging**